ProjectFiles/Assets/Scripts/Game/Core/Driving/VehicleDriver.cs

```diff
@@ -79,5 +84,22 @@ public class VehicleDriver : MonoBehaviour {
         public void setAcceptInput(bool option) {
             acceptinput = option;
         }
+
+        private void OnCollisionEnter(Collision other) {
+            if ((mask.value & 1 << other.gameObject.layer) != 0 && timestamp < Time.time) {
+
+                Rigidbody otherBody = other.gameObject.GetComponent<Rigidbody>();
+
+                float myMomentum    = Vector3.Magnitude(body.velocity)      * body.mass;
+                float otherMomentum = Vector3.Magnitude(otherBody.velocity) * otherBody.mass;
```

Comment:
I feel like this won't work 100% of the time. I think we should resolve the velocity in the direction _between_ the two colliding cars, and do `mass * velocity`, right now it's doing `mass * speed`.

Otherwise, imagine if a car `c1` is travelling in a straight line really fast, and another car `c2` collides perpendicular to it into its side. In the current state, the faster car `c1` will win, even though it's not colliding directly with `c2` no?

```
----------------------------C1------------->
                            ^
                            |
                            |
                            C2

```

(Beautiful diagram)

Reply:
Yeah, you're right. My code would mean that c1 would win out, so long as c2 wasn't much heavier (depending on the speed). I guess it's down to a gameplay perspective since this isn't very realistic anyway. From a gameplay point of view, is it better for c2 to push c1? I think you're probably right and I'm happy to change it either way.

Just so I understand what you mean by velocity between the cars, you're saying the velocity along the vector I have called `colDir`?

Reply:
Yeah I think so. I was also thinking, should we maybe do if the momentum is within some threshold, they both get sent flying? So then we can have head on collisions be wack

Reply:
Yeah that’s probably a good idea, I also think currently, head on collision barely move anyone